### PR TITLE
Feature/migrate material UI v3 v4

### PR DESCRIPTION
**Pull request by Trjegul84**

### CODE REVIEW

File: src/assets/main.css [hunk 1]
```diff
@@ -1405,6 +1426,22 @@ div[id*="panel"] button:hover {
...
+.complist-longer .Select-menu-outer {
+	position: relative !important;
+	height: auto !important;
+	display: block !important;
+}
+
+.Select-option.is-focused  {
+    background-color: rgba(252, 228, 232) !important;
+}
+*/
```
Comment (cazapatamar) [inline src/assets/main.css:1443]: Please delete commented styles

File: src/commons/Selector.jsx [hunk 2]
```diff
@@ -66,13 +66,17 @@ class Selector extends React.Component {
...
+            id="autocomplete-selector"
+            options={data}
+            getOptionLabel={option => option.name}
+            onChange={(event, values) => {
+              handlers[2](parent, values);
             }}
-            name={label || name}
-            data={data}
+            style={{ width: '100%' }}
+            renderInput={params => <TextField {...params} label="Escriba el nombre a buscar" placeholder="Seleccionar..." variant="standard" InputLabelProps={{ shrink: true }} />
```
Comment (cazapatamar) [inline src/commons/Selector.jsx:76]: This line is too long. Use separated lines to put the props of TextField component

**Review by cazapatamar — CHANGES_REQUESTED** [review]

File: src/indicator/ChipManager.jsx [hunk 3]
```diff
@@ -23,14 +23,15 @@ const theme = createMuiTheme({
       },
     },
   },
-  typography: { useNextVariants: true },
+  typography: {
```
Comment (erikasv) [inline src/indicator/ChipManager.jsx:26]: Remove this property since it's empty

**Review by erikasv — CHANGES_REQUESTED** [review]
Also, I think the option list from the autocomplete is too large. Would be better to show only a few options as it is right now in production. Prop `ListboxProps` must help with this.

**Comment by erikasv** [comment]
Also a couple of things missed in the first review:

- The chips in compensations are weird, did you check with Cesar this final result?
- The form to create a new project in compensation is broken (it works, but is ugly and unintelligible)
- The form to add a biome to a new project is broken (it works, but is ugly and unintelligible)
- The selections on section `Donde - Cómo` from compensations are broken (these don't work)

File: src/commons/Selector.jsx (hunk 2 above)
Comment (Trjegul84) [reply src/commons/Selector.jsx]: TextField line was split into columns

File: src/indicator/ChipManager.jsx (hunk 3 above)
Comment (Trjegul84) [reply src/indicator/ChipManager.jsx]: typography was removed.

**Review by cazapatamar — CHANGES_REQUESTED** [review]

File: src/compensation/NewBiomeForm.jsx [hunk 4]
```diff
@@ -15,23 +16,36 @@ class NewBiomeForm extends Component {
...
     const { biomes } = this.props;
     return (
-      <Select
-        ref={this.selectRef}
-        value={biomeSelected}
-        onChange={this.handleChangeBiomes}
-        placeholder="Biomas disponibles"
+      <Autocomplete
+        autoHighlight
         options={biomes.map(biome => ({ ...biome, label: biome.name }))}
```
[the remark this answers is not in the document]
Comment (Trjegul84) [reply src/compensation/NewBiomeForm.jsx]: you are right, this was replaced as you suggest.

**Comment by Trjegul84** [comment]
> Also a couple of things missed in the first review:
> 
>     * The chips in compensations are weird, did you check with Cesar this final result?
> 
>     * The form to create a new project in compensation is broken (it works, but is ugly and unintelligible)
> 
>     * The form to add a biome to a new project is broken (it works, but is ugly and unintelligible)
> 
>     * The selections on section `Donde - Cómo` from compensations are broken (these don't work)

File: src/compensation/NewBiomeForm.jsx [hunk 5]
```diff
@@ -1,8 +1,9 @@
 /** eslint verified */
```
Comment (erikasv) [inline src/compensation/NewBiomeForm.jsx:1]: Please remove this line, it's deprecated now :sweat_smile:

File: src/compensation/NewProjectForm.jsx [hunk 6]
```diff
@@ -1,9 +1,10 @@
 /** eslint verified */
```
Comment (erikasv) [inline src/compensation/NewProjectForm.jsx:1]: Please delete this line

**Review by erikasv — CHANGES_REQUESTED** [review]
In the compensations selector, the autocomplete button style and on hover is different from the one in search, maybe is missing a class or something, see the screenshots

File: src/compensation/NewProjectForm.jsx (hunk 6 above)
Comment (Trjegul84) [reply src/compensation/NewProjectForm.jsx:1]: Described behavior was not found, this is the current view in compensations.
Screenshot_20210201_141833

Comment (erikasv) [reply src/compensation/NewProjectForm.jsx:1]: You need to add more projects to change chips for automplete

File: src/compensation/NewBiomeForm.jsx (hunk 5 above)
Comment (Trjegul84) [reply src/compensation/NewBiomeForm.jsx:1]: Should I remove this /** eslint-env browser */ also ?

Comment (erikasv) [reply src/compensation/NewBiomeForm.jsx:1]: no, that allows us to use no explicitly defined variables (defined by the browser), that line should be only in `index.js` file

Comment (Trjegul84) [reply src/compensation/NewBiomeForm.jsx:1]: Ok. I found 17 fifles with /** eslint verified */ then those must to be removed ?

Comment (erikasv) [reply src/compensation/NewBiomeForm.jsx:1]: No, the idea is to remove it from the files you modified for the issue, not to modify a lot of files only to delete that line.

Comment (Trjegul84) [reply src/compensation/NewBiomeForm.jsx]: Ok, got it.  Line was deleted.

File: src/compensation/NewProjectForm.jsx (hunk 6 above)
Comment (Trjegul84) [reply src/compensation/NewProjectForm.jsx:1]: styles were modified and it looks normal now

Comment (cazapatamar) [reply src/compensation/NewProjectForm.jsx:1]: The autocomplete styles are fine now but when i try to load a project from this component it doesn't load.

image

image

Comment (erikasv) [reply src/compensation/NewProjectForm.jsx:1]: Strange, I don't remember how the list of projects is loaded and what are the conditions for a project to be valid, I would recommend to restore the tables related with projects from production into develop, maybe some project relations are broken.

Btw @Trjegul84 the line this conversation is in hasn't been deleted (src/compensation/NewProjectForm.jsx)

Comment (cazapatamar) [reply src/compensation/NewProjectForm.jsx:1]: > The autocomplete styles are fine now but when i try to load a project from this component it doesn't load.
> 
> image
> 
> image

This bug will be handled in other task and PR

Comment (Trjegul84) [reply src/compensation/NewProjectForm.jsx]: Eslint deprecated command was deleted from NewProjectForm

**Review by erikasv — APPROVED** [review]

**Review by cazapatamar — APPROVED** [review]

**Pull request merged**